### ISSUE TITLE
BEIT: BERT Pre-Training of Image Transformers

### ISSUE THREAD

**Issue by yskim0** [One-Day-One-Paper/One-Day-One-Paper#13]
**BEIT: BERT Pre-Training of Image Transformers**

Hangbo Bao∗, Li Dong, Furu Wei
Microsoft Research

---

# Abstract

- self-supervised vision representation model
- propose a **masked image modeling task**
    - pretrain task
    - 2 views in our pre-training → image patches(16x16), visual tokens(discrete tokens)
- tokenize original image into visual tokens → randomly mask some patches
    - pretraining objective : recover the original **visual tokens** based on the corrupted image patches
- Finetuning on downstream tasks by appending task layers upon the pretrained encoder
    - image classification, semantic segmentation

# Introduction

- why self-supervised ? → data-hungry issue
- SSL ← contrastive learning, self-distillation

![스크린샷 2021-11-10 오전 12.10.02.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/39a6c719-3cdc-4403-a8b0-82d4a33f9176/스크린샷_2021-11-10_오전_12.10.02.png)

- before pre-training, learn "image tokenizer" via autoencoding-style reconstruction (dVAE)
    - image is tokenized into discrete visual tokens according to the learned vocab
- randomly mask some image pathces (gray patches) ⇒ special mask embedding [M]
- pre-training task : predicting the **visual tokens** of the **original image** based on the encoding vectors of the **corrupted image.**
    - instead of the raw pixels of masked patches
    

# Methods

## Image Representations

- 224x224 images into a 14x14 grid of image patches. (patch size 16x16)
- visual token
    - represent the image as **a sequence of discrete tokens** obtained by an "image tokenizer", instead of raw pixels
    - tokenize the image $x \in \mathbb{R}^{H\times W\times C}$ into $z = [z_1,...,z_N] \in \mathcal{V}^{h\times w}$, where the vocabulary $\mathcal{V} = \{1,...,|\mathcal{V}|\}$ contains discrete token indices.
    - vocab size = 8192
- tokenizer
    - learn the image tokenizer via discrete VAE(dVAE)
    - visual token learning을 위해서 2개의 module이 존재
        1. tokenizer
            - $q_\phi(z|x)$ maps image pixels $x$ into discrete tokens $z$ according to a visual codebook(i.e., vocab.)
            - # of visual tokens == # of image patches
        2. decoder
            - $p_\psi(x|z)$ learns to reconstruct the input image $x$ based on the visual tokens $z$
            - reconstruction objective → $\mathbb{E}_{z\sim q_\phi(z|x)}[\log p_\psi(x|z)]$
            - latent visual tokens are discrete → the model traninig is non-differentiable! 
            ⇒ Gumbel-softmax relaxation

## Backbone Network : Image Transformer

- Following ViT

## Pre-Training BEIT : Masked Image Modeling(MIM)

- What to do : randomly mask some % of image patches and then predict the visual tokens that are corresponding to the masked patches
- randomly mask
    - 40% image patches
    - $\{x^p_i\}^N_{i=1}, \{z_i\}^N_{i=1}$
    - masked positions are denoted as $\mathcal{M} \in \{1,...,N\}^{0.4N}$
    - then, replace the masked patches with a learnable embedding $e_{[M]} \in \mathbb{R}^D$
- corrupted image patches
    - $x^{\mathcal{M}} = \{x^p_i:i\notin \mathcal{M} \}^N_{i=1} \cup \{e_{[M]} : i \in \mathcal{M}\}^N_{i=1}$
    - fed into the L-layer Transformer
    - final hidden vectors $\{h^L_i\}^N_{i=1}$ are regarded as encoded representations of the input patches
- predict the corresponding visual tokens
    - $p_{MIM}(z'|x^\mathcal{M})=softmax_{z'}(W_ch_i^L+b_c)$
    - pre-training objective is to maximize the log-likelihood of the correct visual tokens $z_i$ given the corrupted image
    - $max \sum_{x\in \mathcal{D}} \mathbb{E}_\mathcal{M} [\sum_{i\in \mathcal{M}} \log p_{MIM}(z_i|x^\mathcal{M})$
        - D → training corpus
        - M → randomly masked positions
        - x^M → corrupted image
- Blockwise Masking
    
    ![스크린샷 2021-11-10 오전 12.37.48.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/9607f1a8-4fc8-42f6-b276-c35990ba85dd/스크린샷_2021-11-10_오전_12.37.48.png)
    
- pixel-level auto-encoding for vision pre-training → pushes the model to focus on short-range dependencies and high-frequency details
- BEIT : predicting discrete visual tokens which summarizes the details to high-level abstracitons

## Variational Autoencoder

- $x$ : original image
$\tilde{x}$ : masked image
$z$  : visual tokens

![스크린샷 2021-11-10 오전 12.41.07.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/2991d935-ba7f-4d79-8a7a-443062e9e398/스크린샷_2021-11-10_오전_12.41.07.png)

- $q_\phi(z|x)$ : the image tokenizer that obtains visual tokens
- $p_\psi(x|z)$  : decodes the original image given input visual tokens
- $p_\theta(z|\tilde{x})$ : recovers the visual tokens based on the masked image (MIM pre-training task)

![스크린샷 2021-11-10 오전 12.41.16.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/261cb5ef-653c-4b8b-a576-4ae1324314c8/스크린샷_2021-11-10_오전_12.41.16.png)

## Fine-tuning on Downstream Vision Tasks

- image classification
    - category probs. → $softmax(avg(\{h_i^L\}^L_{i=1}W_c))$
    - maximize the likelihood of labeled data by updating the params. of BEIT and the softmax classifier
- Semantic segmentation

# Experiments

![스크린샷 2021-11-10 오전 12.45.29.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/fa18fe84-8148-4cad-85bd-81e651714b4b/스크린샷_2021-11-10_오전_12.45.29.png)

![스크린샷 2021-11-10 오전 12.45.50.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/329a68fa-3730-46f3-bdb3-14ca15402a30/스크린샷_2021-11-10_오전_12.45.50.png)

![스크린샷 2021-11-10 오전 12.46.20.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/13458f06-ca4c-4b8d-88cd-2224f7711e5e/스크린샷_2021-11-10_오전_12.46.20.png)

# Conclusion

- BERT-like pre-training(i.e., auto-encoding with masked input)
- achieves strong fine-tuning on downstream tasks
